### ISSUE TITLE
Support for triggering a CI rebuild without junk commits

### ISSUE THREAD

**Issue by iandees** [openaddresses/machine#378]
Currently, the only way a build can be triggered by machine CI is via the webhook from GitHub on a new commit. It'd be nice if we could click a button on the CI results page or add a comment on the pull request to trigger a rebuild.

**Comment by migurski** [comment]
There are three places I could see this being useful:

* Attached to a single source in an individual job, e.g. http://results.openaddresses.io/jobs/b3086ac6-e30c-4691-994b-cc312ee70b22
* Next to each source on the front page at https://results.openaddresses.io
* Next to each source on a set page, e.g. http://results.openaddresses.io/sets/106335/

Would require Github OAuth authentication for visitors, I think.

**Comment by migurski** [comment]
I’ve made some steps in this direction, culminating in a basic Github login button at https://results.openaddresses.io/auth. You can log in if you're part of the OA org here. 

How should I prioritize making this work? The three cases above aren't compatible with getting [this PR into the queue](https://github.com/openaddresses/openaddresses/pull/2044), for example. I have been thinking to instead create a submission form where Github URLs could be copy/pasted. 

Help?

**Comment by iandees** [comment]
One option would be to not require auth at all. Just have the link kick off a re-cache regardless of who found the button to click (except maybe a robot via robots.txt).

Another option would be to listen to comments on the pull request. For example, the Jenkins pull request builder rebuilds the pull request if you post "retest this please". The [comment webhook](https://developer.github.com/v3/orgs/hooks/) includes the user ID and we could whitelist only certain IDs to trigger builds on?

**Comment by migurski** [comment]
Comments, yeah. That puts the action a little closer to the right place, though it also has a slight "magic" feel to it. Any whitelist would be composed of organization members, I think. Do you know if the Jenkins rebuilder puts some kind of feedback into the thread?

**Comment by iandees** [comment]
I don't think the Jenkins builder adds a comment, it just triggers a build. Usually what I would do is make the comment and then delete it. It's pretty hacky, but it helps with auth problems at least.

**Comment by migurski** [comment]
Do you mean, you as an end-user would delete your own comment after the build happened?

**Comment by iandees** [comment]
Yep. I would delete the comment I made.

**Comment by migurski** [comment]
Another question, tangential-but-pertinent to this one: authentication tokens for updating statuses in pull requests require a personal account, not an organizational one. For OA, we're using one of mine. I’m slightly uncomfortable about this, because it’s an invisible dependency. Might we want to have a dedicated bot account for this purpose?

**Comment by iandees** [comment]
Yep, we can create a bot account. It'd be nicer if GitHub would let the app request access to only one organization instead of all of 'em...

**Comment by migurski** [comment]
In the meantime, I've swapped out the configured token for one that's authenticated against the CI app, so now the OA icon shows up in statuses instead of my personal account icon.

**Comment by migurski** [comment]
Hm, the status POST back to Github failed for reasons I don’t understand, though it tried to do the right thing:

```
 File "/usr/local/lib/python2.7/dist-packages/openaddr/ci/webhooks.py", line 106, in app_hook
   gag_status)
 File "/usr/local/lib/python2.7/dist-packages/openaddr/ci/__init__.py", line 121, in process_github_payload
   update_pending_status(status_url, job_url, filenames, github_auth)
 File "/usr/local/lib/python2.7/dist-packages/openaddr/ci/__init__.py", line 409, in update_pending_status
   return post_github_status(status_url, status, github_auth)
 File "/usr/local/lib/python2.7/dist-packages/openaddr/ci/__init__.py", line 397, in post_github_status
   raise ValueError('Failed status post to {}'.format(status_url))
ValueError: Failed status post to https://api.github.com/repos/openaddresses/openaddresses/statuses/4f041e83ab26b82652152bbfcefde0e550639aef
```

I will investigate.

**Comment by migurski** [comment]
Looks like the status JSON `target_url` was not being properly set to an absolute URL, so I’ve messily patched that in 2.33.4. Going to go for a better, more-correct fix.

**Comment by migurski** [comment]
When a comment is left on a pull request with a line that starts with "Rerun this please", then a cacheless rebuild will be triggered. 

Confirmed in https://github.com/openaddresses/openaddresses/pull/2044 that this works.
